sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java

```diff
@@ -101,14 +120,35 @@ public UploadManifestResult uploadManifest(BinaryData data) {
      * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
      *
      * @param data The manifest that needs to be uploaded.
+     * @param options The options for the upload manifest operation.
+     * @return The operation result.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code data} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public UploadManifestResult uploadManifest(BinaryData data, UploadManifestOptions options) {
+        return this.asyncClient.uploadManifest(data, options).block();
+    }
+
+    /**
+     * Uploads a manifest to the repository.
+     * The client currently only supports uploading OciManifests to the repository.
+     * And this operation makes the assumption that the data provided is a valid OCI manifest.
+     * <p>
+     * Also, the data is read into memory and then an upload operation is performed as a single operation.
+     *
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     *
+     * @param data The manifest that needs to be uploaded.
+     * @param options The options for the upload manifest operation.
      * @param context Additional context that is passed through the Http pipeline during the service call.
      * @return The rest response containing the operation result.
      * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
      * @throws NullPointerException thrown if the {@code data} is null.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Response<UploadManifestResult> uploadManifestWithResponse(BinaryData data, Context context) {
-        return this.asyncClient.uploadManifestWithResponse(data.toByteBuffer(), context).block();
+    public Response<UploadManifestResult> uploadManifestWithResponse(BinaryData data, UploadManifestOptions options, Context context) {
```

Comment:
Generally, if options exist they take all parameters aside from Context, so data should be merged into options.

https://azure.github.io/azure-sdk/java_introduction.html#java-params-complex-overloads

Reply:
That is a good point. I may need to circle back with the FC on this one. I believe it was added because we do not fully understand all the 'options' we may need to add and the type gives us that flexibility.

But yes, I will update this and the other overloads :)